src/target/samd.c

```diff
@@ -129,15 +130,15 @@ const command_s samd_cmd_list[] = {
 #define SAMD_STATUSB_PROT    (1U << 16U)
 
 /* Device Identification Register (DID) */
-#define SAMD_DID_MASK          0xff380000U
+#define SAMD_DID_MASK          0xfe380000U
 #define SAMD_DID_CONST_VALUE   0x10000000U
 #define SAMD_DID_DEVSEL_MASK   0xffU
 #define SAMD_DID_DEVSEL_POS    0U
 #define SAMD_DID_REVISION_MASK 0x0fU
 #define SAMD_DID_REVISION_POS  8U
 #define SAMD_DID_SERIES_MASK   0x1fU
 #define SAMD_DID_SERIES_POS    16U
-#define SAMD_DID_FAMILY_MASK   0x3fU
+#define SAMD_DID_FAMILY_MASK   0x1fU
```

Comment:
Interesting, looking at #571 and https://ww1.microchip.com/downloads/aemDocuments/documents/MCU32/ProductDocuments/DataSheets/SAM-D21-DA1-Family-Data-Sheet-DS40001882H.pdf this undoes a fix that we introduced in #1162 where the mask 0x1f is insufficient - but for the series rather than the family as the fix applied to the wrong define by accident. See §13.13.9 Device Identification of the linked datasheet.

Please make SAMD_DID_SERIES_MASK 0x3f to correct the fix and apply this switch-over of the mask in a separate commit to the rest of this.